snuba/datasets/cdc/groupassignee.py

```diff
@@ -79,14 +80,14 @@ def __init__(self) -> None:
                 ),
                 postgres_table=self.POSTGRES_TABLE,
             ),
-            query_processors=[PrewhereProcessor()],
+            query_processors=[],
         )
 
-        storage_selector = SingleStorageSelector(storage=storage)
-
         super().__init__(
             storages=[storage],
-            storage_selector=storage_selector,
+            query_plan_builder=SingleTableQueryPlanBuilder(
+                storage=storage, post_processors=[PrewhereProcessor()],
+            ),
```

Comment:
The issue we have with adding the prewhere processing in the individual storage is that it may not work for joins.
Let's say we have a join between two storages (since one storage is one table), each of them has its own prewhere candidates. We could execute the prewhere processor individually, and each one would move some conditions from the where clause to the pre-where.
In a join, we want to apply the pre-where conditions only to the most external table (I suspect it would not even work otherwise) and in general we want to cap the number of conditions we add to the prewhere clause.
This means the prewhere clause cannot be fully processed at storage level in isolation. 
What we could do is to apply that processing on the individual storages, and then reprocessing the clause in the planner to adjust it or remove individual conditions.

Right now it would work either way since the PrewhereProcessor takes the list of candidates from the query `data source` which is populated after the storage is selected, so it would work on SelectedTableQueryPlanBuilder as well (https://github.com/getsentry/snuba/blob/master/snuba/query/processors/prewhere.py#L26).

I am open to move this back to the storage and add a Prewhere-re-processor in the planner to adjust what the storages have decided (which essentially would do nothing in all cases but joins). What do you think.